pkg/cluster/sync.go

```diff
@@ -556,7 +556,13 @@ func (c *Cluster) syncRoles() (err error) {
 
 	// create list of database roles to query
 	for _, u := range c.pgUsers {
-		userNames = append(userNames, u.Name)
+		pgRole := u.Name
+		if u.Namespace != c.Namespace {
```

Comment:
make sure `u.Namespace` is always set. Currently it's missing in `initTeamMembers` and `readPgUsersFromDatabase` methods which leads to the wrong name appended here. This makes the second e2e test fail atm (it wants to create a user that already exists).